src/include/sof/audio/component.h

```diff
@@ -702,11 +702,20 @@ int comp_set_state(struct comp_dev *dev, int cmd);
 static inline void component_set_period_frames(struct comp_dev *current,
 					       uint32_t rate)
```

Comment:
@RanderWang how did you test this? afaik all topologies use 48Khz as a rate. 
Did you also changed the topologies?

We will need in the future to make setting stream params more flexibile (e.g: rate, format, channels). So far most of them are hardcoded in the topology.

Reply:
yes, I made a topology with 44.1khz enabled